src/drivers/imx/sai.c

```diff
@@ -424,7 +424,11 @@ static int sai_get_hw_params(struct dai *dai,
 
 	/* SAI only currently supports these parameters */
 	params->rate = sai->params.fsync_rate;
+#ifdef CONFIG_IMX8ULP
+	params->channels = 1;
+#else
 	params->channels = 2;
+#endif
 	params->buffer_fmt = 0;
```

Comment:
I'm fine with this for now. In the future the number of channels should be read from sai->params.